appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -81,7 +86,9 @@ async function getLatestDeployment(kuduClient: KuduClient, permanentId: string |
     let deployment: DeployResult | undefined;
     if (permanentId) {
         // Use "permanentId" to find the deployment during its "permanent" phase
-        deployment = await kuduClient.deployment.getResult(permanentId);
+        // if we have a "permanentId" we know which deployment we are looking for, so wait until latest id reflects that
+        const latestDeployment: DeployResult = await kuduClient.deployment.getResult('latest');
+        deployment = latestDeployment.id === permanentId ? latestDeployment : undefined;
     } else if (initialStartTime) {
         // Use "initialReceivedTime" to find the deployment during its "temp" phase
         deployment = (await kuduClient.deployment.getDeployResults())
```

Comment:
Can you refresh my memory on why the existing `latest` and `initialStartTime` logic doesn't work? Your change is definitely an improvement and I like it, but we're still going to miss the logs during the "temp" phase (although I don't remember what happens in the temp phase or how long it takes)

Reply:
I think this is what you're asking:

So the reason `Redeploy` and `Local Git` deployment don't work with old code is due to a timing issue.  Both methods can't be `await`ed in order for the logs to stream in real-time because the API call doesn't resolve _until_ after the deployment has completed.  But due to the Kudu taking some time to trigger the new build, our old code would grab the "latest" (which would be the previous deployment) and just return all of those logs immediately.

Reply:
Okay yep I remember now